gdnative-derive/src/derive_macro.rs

```diff
@@ -26,6 +27,19 @@ pub(crate) fn parse_derive_input(input: TokenStream) -> DeriveData {
     let base = syn::parse::<Type>(inherit_attr.tts.clone().into())
         .expect("`inherits` attribute requires the base type as an argument.");
 
+    let user_data = input
+        .attrs
+        .iter()
+        .find(|a| a.path.segments[0].ident == "user_data")
+        .map(|attr| {
+            syn::parse::<Type>(attr.tts.clone().into())
+                .expect("`userdata` attribute requires a type as an argument.")
+        })
+        .unwrap_or_else(|| {
+            syn::parse::<Type>(quote! { ::gdnative::user_data::MutexData<#ident> }.into())
```

Comment:
Is there a way we could make this default configurable as a constant? If not that's fine, might just be nicer in case we want to add feature flags for overriding defaults or disabling "threading support" in future. It's not a must, just a suggestion which does not block this PR :)

Reply:
I think we can have a type alias for that.